### PR TITLE
Mobile table v2

### PR DESCRIPTION
**Pull request by youngfreezyVA**
here it is in action:
https://drive.google.com/file/d/1GwDAfr5TvHeid0t_DuNp4hKQMpL4iruH/view?usp=sharing

https://github.com/department-of-veterans-affairs/va.gov-team/issues/15375

### CODE REVIEW

**Comment by kelsonic** [comment]
Could you ping us in a comment here when it's all ready for review? 🙂

**Review by andaleliz — COMMENTED** [review]
:clapping: looks great! The only thing I see in the video is on the desktop version - the cell data should be top aligned instead of center/middle aligned. 

I didn't review the CSS, focused on the look and feel from the video.

**Comment by youngfreezyVA** [comment]
@kelsonic @jenstrickland @ncksllvn this is ready for review!
 @andaleliz thanks for taking a look and the feedback! I didn't touch the desktop headers :) didn't change the html markup for desktop, and all my css changes are wrapped in a media query.  if that's an issue we can create a separate ticket for it and i'm happy to take that one on

File: src/site/paragraphs/table.drupal.liquid
Comment (kelsonic) [inline src/site/paragraphs/table.drupal.liquid:45]: Won't this make it so that if there is more than length 2 for `entity.fieldTable.value`, the middle values won't be rendered?

Comment (kelsonic) [reply src/site/paragraphs/table.drupal.liquid:45]: Wondering if that should be an `else` instead?

Comment (kelsonic) [reply src/site/paragraphs/table.drupal.liquid:45]: Oh nevermind seeing now that this is how it was previously, so better not to mess with that lol

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Nice work and perseverance with this one!

File: src/site/paragraphs/table.drupal.liquid
Comment (ncksllvn) [inline src/site/paragraphs/table.drupal.liquid:50]: I haven't seen `forloop.index0` before. Are we just trying to get the first element out of `colLabels`?

File: src/site/filters/liquid.js
Comment (ncksllvn) [inline src/site/filters/liquid.js:373]: There is already a [`get`](https://github.com/leizongmin/tinyliquid/blob/cebcb26c3839e725cf0469dccc0073799902a020/README_en.md#arrays-and-objects) filter that might do what we need? It works to get dynamic keys out of object, thinking it might work for array indexes as well?

File: src/platform/site-wide/sass/modules/_m-responsive-table.scss
Comment (ncksllvn) [inline src/platform/site-wide/sass/modules/_m-responsive-table.scss:49]: What are your thoughts on `va-table--responsive` instead of `rwd`? 

I'm also wondering on the "3" in `td:nth-child(3)` - what that numbers means versus a 2 or 4

**Comment by ncksllvn** [comment]
@youngfreezyVA Thank you for the fantastic video btw!

File: src/site/paragraphs/table.drupal.liquid
Comment (youngfreezyVA) [reply src/site/paragraphs/table.drupal.liquid:45]: hehe agreed :)

File: src/site/paragraphs/table.drupal.liquid
Comment (youngfreezyVA) [reply src/site/paragraphs/table.drupal.liquid:50]: `index0` is zero based iteration whereas `index` starts off at 1... smh drupal :P
we are trying to get the associated col label (From what used to be the table header, which is hidden in mobile) for each column value

File: src/site/filters/liquid.js
Comment (youngfreezyVA) [reply src/site/filters/liquid.js:373]: let me check it out...

Comment (youngfreezyVA) [reply src/site/filters/liquid.js:373]: you are correct ... `getValueFromObjPath` works for arrays as well... so I added in a comment above `getValueFromObjPath` to indicate it works for arrays also

Comment (ncksllvn) [reply src/site/filters/liquid.js]: Oh cool, thank you 😄

File: src/site/paragraphs/table.drupal.liquid
Comment (ncksllvn) [reply src/site/paragraphs/table.drupal.liquid]: Wow yeah, I see it in the Shopify docs now, https://shopify.dev/docs/themes/liquid/reference/objects/for-loops. Thanks for the insight!

File: src/platform/site-wide/sass/modules/_m-responsive-table.scss
Comment (youngfreezyVA) [reply src/platform/site-wide/sass/modules/_m-responsive-table.scss:49]: i tried to add in va-table--responsive instead of rwd and it broke everything 😬
the `rwd` class is from jens codepen
re: the specific selector you pointed out.. it wasn't doing anything, so i removed it as well as a couple of other selectors that were specific to the structure in the codepen and didn't match the data returned from the cms.

Comment (ncksllvn) [reply src/platform/site-wide/sass/modules/_m-responsive-table.scss]: Oh I just meant if we could use something like `va-table--responsive ` as the class name instead of `rwd`. Off the bat, I don't know what `rwd` stands for and just thinking we could use a more descriptive name

Comment (youngfreezyVA) [reply src/platform/site-wide/sass/modules/_m-responsive-table.scss]: ah gotcha... that makes sense - done.

Comment (jenstrickland) [reply src/platform/site-wide/sass/modules/_m-responsive-table.scss]: Yes! Y'all should definitely change `.rwd` to use the naming convention used on VA.gov. I used `.rwd` in mine because I don't know what y'all would use (rwd stands for responsive web design, but it's totally my shorthand sketch).

File: src/platform/site-wide/sass/modules/_m-responsive-table.scss
Comment (ncksllvn) [inline src/platform/site-wide/sass/modules/_m-responsive-table.scss:52]: This is looking great. I'm wondering about the `td` rules and the `.column-value` rules. Should they also be prefaced by `.va-table--responsive`? 

Your call, but we could just wrap the whole thing ->

```scss
.va-table--responsive {
  tr {
    border-top: 2px solid #999;
  }
  thead {
    display: none;
  }
  td,
  th {
    border: none;
    display: block;
  }
  td {
    display: flex;
    grid-gap: 2px;
    padding: 1rem;
    padding-left: 0px;
  }
  dfn {
    display: block;
    font-style: normal;
    font-weight: 800;
    margin: 0;
    padding: 0;
  }
  td p {
    margin-bottom: 0;
  }
  .column-value strong:nth-child(1) {
    font-weight: normal;
  }
  .column-label:nth-child(1) {
    padding-top: 20px;
  }
  .column-value:nth-last-child(1) {
    padding-bottom: 20px;
  }
}
```

Comment (youngfreezyVA) [reply src/platform/site-wide/sass/modules/_m-responsive-table.scss:54]: 💯 yes i thought to do the same thing but forgot. good call

**Review by ncksllvn — APPROVED** [review]
LGTM! Thanks for making the updates and so quickly! Just please test one last time before merge to make sure everything is 👌

**Comment by youngfreezyVA** [comment]
> LGTM! Thanks for making the updates and so quickly! Just please test one last time before merge to make sure everything is 👌

yup its looking like everything is working as expected @ncksllvn ... do i need to wait on anyone else to merge this in?

**Comment by ncksllvn** [comment]
@youngfreezyVA Yeah I suppose we do need to wait...If you look at the `Code owner review required` we can see that the frontend-review-group (this is the platform team, aka VSP) still needs to sign off

File: src/site/paragraphs/table.drupal.liquid
Comment (billfienberg) [inline src/site/paragraphs/table.drupal.liquid:38]: I think rendering the `thead` as a child of `tbody` is technically invalid HTML. 

>4.9.1 The `table` element - Content Model: In this order: optionally a `caption` element, followed by zero or more `colgroup` elements, followed optionally by a `thead` element, followed by either zero or more `tbody` elements or one or more `tr` elements, followed optionally by a `tfoot` element, optionally intermixed with one or more script-supporting elements. - https://html.spec.whatwg.org/multipage/tables.html#the-table-element

>4.9.5 The `tbody` element - Content Model: Zero or more `tr` and script-supporting elements. - https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element

File: src/site/paragraphs/table.drupal.liquid
Comment (billfienberg) [inline src/site/paragraphs/table.drupal.liquid:49]: I think rendering the `dfn` as an immediate child of `tr` is technically invalid HTML. 

>**4.9.8 The `tr` element - Content Model**: Zero or more `td`, `th`, and script-supporting elements. - https://html.spec.whatwg.org/multipage/tables.html#the-tr-element

I think it would be valid if you moved the `dfn` to be a child of the `td` instead of being a child of the `tr`.

Comment (youngfreezyVA) [reply src/site/paragraphs/table.drupal.liquid]: done

File: src/site/paragraphs/table.drupal.liquid
Comment (youngfreezyVA) [reply src/site/paragraphs/table.drupal.liquid:37]: addressed this feedback as well :)

**Comment by youngfreezyVA** [comment]
@billfienberg @kelsonic @ncksllvn @andaleliz 
addressed all feedback and here's the responsive table in action, post pr feedback:
https://drive.google.com/file/d/1hoaMSfFHFiho6KosKns78K9tL6EMnKHN/view?usp=sharing

**Review by billfienberg — APPROVED** [review]

**Pull request merged**
